types/tx.go

```diff
@@ -1,7 +1,43 @@
 package types
 
+import (
+	"github.com/tendermint/tendermint/crypto/merkle"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+)
+
 // Tx represents transactoin.
 type Tx []byte
 
 // Txs represents a slice of transactions.
 type Txs []Tx
+
+// Hash computes the TMHASH hash of the wire encoded transaction.
+func (tx Tx) Hash() []byte {
+	return tmhash.Sum(tx)
+}
+
+// Proof returns a simple merkle proof for this node.
+// Panics if i < 0 or i >= len(txs)
+// TODO: optimize this!
+func (txs Txs) Proof(i int) TxProof {
```

Comment:
I took the implementation in this file from celestia-core repo (types/tx.go)

@tzdybal I was not sure if those functions needed to be implemented in Optimint or if the typings needed to change in the Optimint repo to use `"github.com/tendermint/tendermint/types"`

For example, should `Txs` type in `types/block.go` (Optimint repo) be using the tendermind typings?

Reply:
For now, I think your solution is most pragmatic.
`Txs` types are very problematic; I would love not to convert between them in so many places. But I think this will be a part of bigger refactoring of types system, related to https://github.com/celestiaorg/optimint/issues/37.

Reply:
Should we replace the tendermint with celestia-core? I see we do [this](https://github.com/celestiaorg/celestia-app/blob/master/go.mod#L129) in celestia-app.

Reply:
No. This is fine. Optimint is intended as tendermint replacement for plain cosmos-sdk apps. It's better to introduce few lines of code than dependency,